multi-node/aws/pkg/cluster/cluster.go

```diff
@@ -42,38 +43,104 @@ func New(cfg *config.Cluster, awsDebug bool) *Cluster {
 	}
 
 	return &Cluster{
-		clusterName: aws.String(cfg.ClusterName),
-		svc:         cloudformation.New(session.New(awsConfig)),
+		Cluster: *cfg,
+		session: session.New(awsConfig),
 	}
 }
 
 type Cluster struct {
-	clusterName *string
-	svc         *cloudformation.CloudFormation
+	config.Cluster
+	session *session.Session
 }
 
 func (c *Cluster) ValidateStack(stackBody string) (string, error) {
-	input := &cloudformation.ValidateTemplateInput{
+	validateInput := cloudformation.ValidateTemplateInput{
 		TemplateBody: &stackBody,
 	}
 
-	validationReport, err := c.svc.ValidateTemplate(input)
+	cfSvc := cloudformation.New(c.session)
+	validationReport, err := cfSvc.ValidateTemplate(&validateInput)
 	if err != nil {
-		return "", fmt.Errorf("Invalid cloudformation stack: %v", err)
+		return "", fmt.Errorf("invalid cloudformation stack: %v", err)
 	}
 
-	return validationReport.String(), err
+	return validationReport.String(), nil
+}
+
+type ec2Service interface {
+	DescribeVpcs(*ec2.DescribeVpcsInput) (*ec2.DescribeVpcsOutput, error)
+	DescribeSubnets(*ec2.DescribeSubnetsInput) (*ec2.DescribeSubnetsOutput, error)
+}
+
+func (c *Cluster) validateExistingVPCState(ec2Svc ec2Service) error {
+	if c.VPCID == "" {
+		//The VPC will be created. No existing state to validate
+		return nil
+	}
+
+	describeVpcsInput := ec2.DescribeVpcsInput{
+		VpcIds: []*string{aws.String(c.VPCID)},
+	}
+	vpcOutput, err := ec2Svc.DescribeVpcs(&describeVpcsInput)
+	if err != nil {
+		return fmt.Errorf("error describing existing vpc: %v", err)
+	}
+	if len(vpcOutput.Vpcs) == 0 {
+		return fmt.Errorf("could not find vpc %s in region %s", c.VPCID, c.Region)
+	}
+	if len(vpcOutput.Vpcs) > 1 {
+		//Theoretically this should never happen. If it does, we probably want to know.
+		return fmt.Errorf("found more than one vpc with id %s. this is NOT NORMAL.", c.VPCID)
+	}
+
+	existingVPC := vpcOutput.Vpcs[0]
+
+	if *existingVPC.CidrBlock != c.VPCCIDR {
+		//If this is the case, our network config validation cannot be trusted and we must abort
+		return fmt.Errorf("configured vpcCidr (%s) does not match actual existing vpc cidr (%s)", c.VPCCIDR, *existingVPC.CidrBlock)
+	}
+
+	describeSubnetsInput := ec2.DescribeSubnetsInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("vpc-id"),
+				Values: []*string{existingVPC.VpcId},
+			},
+		},
+	}
+
+	subnetOutput, err := ec2Svc.DescribeSubnets(&describeSubnetsInput)
+	if err != nil {
+		return fmt.Errorf("error describing subnets for vpc: %v", err)
+	}
+
+	subnetCIDRS := make([]string, len(subnetOutput.Subnets))
+	for i, existingSubnet := range subnetOutput.Subnets {
+		subnetCIDRS[i] = *existingSubnet.CidrBlock
+	}
+
+	if err := c.ValidateExistingVPC(*existingVPC.CidrBlock, subnetCIDRS); err != nil {
```

Comment:
Up to you if you want to change this:

I had suggested breaking this out to make testing possibly easier, but the way you have it mocked out, it's not really necessary (and this function ends up only being called here).

You could inline this function again in that case, if it seemed simpler than creating temp vars to pass into this. But again, totally up to you (just noticed my original suggestion doesn't really help in this case).